source/patterns/@aws-solutions-constructs/aws-iot-s3/lib/index.ts

```diff
@@ -112,7 +110,7 @@ export class IotToS3 extends Construct {
     this.iotTopicRule = new iot.CfnTopicRule(this, 'IotTopicRule', iotTopicProps);
 
     // If bucket has a KMS CMK, explicitly provide IoTActionsRole necessary access to write to the bucket
-    if (this.s3Bucket && this.s3Bucket.encryptionKey) {
+    if (this.s3Bucket && this.s3Bucket.encryptionKey && props.existingBucketObj) {
```

Comment:
I don't get this - doesn't the role need access to the KMS key for the bucket regardless of whether the bucket is new or was supplied by the client?

Reply:
default encryption for new buckets is SSE-S3, and it doesn't need explicit permissions